src/ripple/rpc/handlers/HealthCheck.cpp

```diff
@@ -0,0 +1,119 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2014 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/misc/NetworkOPs.h>
+#include <ripple/json/json_value.h>
+#include <ripple/net/RPCErr.h>
+#include <ripple/protocol/jss.h>
+#include <ripple/rpc/Context.h>
+#include <ripple/rpc/impl/TransactionSign.h>
+#include <ripple/rpc/Role.h>
+
+namespace ripple {
+
+Json::Value
+doHealthCheck(RPC::JsonContext& context)
+{
+    bool constexpr humanReadable = true;
+    bool constexpr noAdmin = false;
+    bool constexpr noCounters = false;
+    auto info = context.netOps.getServerInfo(
+                    humanReadable, noAdmin, noCounters);
+
+    int last_validated_ledger_age = std::numeric_limits<int>::max();
+    if (info.isMember("validated_ledger"))
+        last_validated_ledger_age = info["validated_ledger"]["age"].asInt();
+    bool amendment_blocked = false;
+    if (info.isMember("amendment_blocked"))
+        amendment_blocked = true;
+    int number_peers = info["peers"].asInt();
+    std::string server_state = info["server_state"].asString();
+    auto load_factor = info["load_factor"].asDouble();
+
+    Json::Value ret = Json::objectValue;
+    enum {healthy, warning, critical};
+    int health = healthy;
+    auto set_health = [&health](int state)
+        {
+            if (health < state)
+                health = state;
+        };
+
+    if (last_validated_ledger_age >= 7)
```

Comment:
I'm curious as to the rationality that went into these static values, after all since a ledger currently closes every 2-4 seconds, doesn't 7 seconds seem to be a bit excessive / long to report as being healthy? Was this based on data collected from actual propagation times to read-only nodes and such?

Also since close time is variable and changes on a ledger by ledger basis, and the algorithm itself is subject to change (see: #2958) does it make sense to have these hard coded values in this rpc method? Furthermore node operators leverage hardware w/ different performance capabilities may want to map different performance levels to the healthy, warning, critical states. Thoughts on moving these static / hard coded values to the config file or similar?